subsys/net/ip/net_context.c

```diff
@@ -1571,6 +1570,11 @@ static int context_sendto(struct net_context *context,
 		}
 	}
 
+	iface = net_context_get_iface(context);
```

Comment:
Should this be moved to the top of this function, like line 1410? In case the interface is down, it would be detected earlier and less unnecessary code is executed.

Reply:
The code does various sanitychecks of the destination address etc. so I think we should do those first, and only when all the params are correctly set and we are about to start to allocate the net_pkt, we verify that we can actually send stuff.